### DISCUSSION TITLE
Problem with 2 Smart Meters ond RS485

### DISCUSSION THREAD

**Discussion by spoeh-man** [arendst/Tasmota#18007] [Q&A] [answered]
I Have 2 Smart meters from ABB B23 (MODBus) i used this script

>D
>B
->sensor53 r
>M 1
+1,3,m,0,9600,ABB,1,10,01035B00,01035B02,01035B04,01035B14,01035B16,01035B18,01035B1A,r010350080004,r010350000004,r010350040004
1,010304UUuuUUuu@i0:10,Voltage L1-N,V,Voltage_L1,1
1,010304UUuuUUuu@i1:10,Voltage L2-N,V,Voltage_L2,1
1,010304UUuuUUuu@i2:10,Voltage L3-N,V,Voltage_L3,1
1,010304SSssSSss@i3:100,Active power Total,W,Active_power_Total,2
1,010304SSssSSss@i4:100,Active power L1-N,W,Active_power_L1,2
1,010304SSssSSss@i5:100,Active power L2-N,W,Active_power_L2,2
1,010304SSssSSss@i6:100,Active power L3-N,W,Active_power_L3,2
1,010308xxxxxxxxSSssSSss@i7:100,Real energy,kWh,Real_energy,2
1,010308xxxxxxxxUUuuUUuu@i8:100,Real energy consumed,kWh,Real_energy_consumed,2
1,010308xxxxxxxxUUuuUUuu@i9:100,Real energy delivered,kWh,Real_energy_delivered,2
#

For One Meter with Device id 1 it will work but now i will do this for Two meters one with adress 1 and the other with adresss 2 can anybody check is this right?

>D
>B
->sensor53 r
>M 2
+1,3,m,0,9600,Wohnen,1,10,01035B00,01035B02,01035B04,01035B14,01035B16,01035B18,01035B1A,r010350080004,r010350000004,r010350040004
1,010304UUuuUUuu@i0:10,Voltage L1-N,V,Voltage_L1,1
1,010304UUuuUUuu@i1:10,Voltage L2-N,V,Voltage_L2,1
1,010304UUuuUUuu@i2:10,Voltage L3-N,V,Voltage_L3,1
1,010304SSssSSss@i3:100,Active power Total,W,Active_power_Total,2
1,010304SSssSSss@i4:100,Active power L1-N,W,Active_power_L1,2
1,010304SSssSSss@i5:100,Active power L2-N,W,Active_power_L2,2
1,010304SSssSSss@i6:100,Active power L3-N,W,Active_power_L3,2
1,010308xxxxxxxxSSssSSss@i7:100,Real energy,kWh,Real_energy,2
1,010308xxxxxxxxUUuuUUuu@i8:100,Real energy consumed,kWh,Real_energy_consumed,2
1,010308xxxxxxxxUUuuUUuu@i9:100,Real energy delivered,kWh,Real_energy_delivered,2
+2,3,m,0,9600,Geschäft,1,10,02035B00,02035B02,02035B04,02035B14,02035B16,02035B18,02035B1A,r020350080004,r020350000004,r020350040004
2,020304UUuuUUuu@i0:10,Voltage L1-N,V,Voltage_L1,1
2,020304UUuuUUuu@i1:10,Voltage L2-N,V,Voltage_L2,1
2,020304UUuuUUuu@i2:10,Voltage L3-N,V,Voltage_L3,1
2,020304SSssSSss@i3:100,Active power Total,W,Active_power_Total,2
2,020304SSssSSss@i4:100,Active power L1-N,W,Active_power_L1,2
2,020304SSssSSss@i5:100,Active power L2-N,W,Active_power_L2,2
2,020304SSssSSss@i6:100,Active power L3-N,W,Active_power_L3,2
2,020308xxxxxxxxSSssSSss@i7:100,Real energy,kWh,Real_energy,2
2,020308xxxxxxxxUUuuUUuu@i8:100,Real energy consumed,kWh,Real_energy_consumed,2
2,020308xxxxxxxxUUuuUUuu@i9:100,Real energy delivered,kWh,Real_energy_delivered,2
#
Sorry for my bad english im from Germany

**Comment by gemu2015** [comment]
there are examples in the sml docs how more than 1 MODBUS meters are handled.
2 meters on the same bus are treated as 1 meter, only the request codes are different
 so remove the 2. meter definition and copy the request codes of the 2. meter behind the first one
also correct the indices to be incremented with each request

**Comment by spoeh-man** [comment]
Like This ?
Sorry my englisch is not that good to understand all the examples

D
B
->sensor53 r
M 1
+1,3,m,0,9600,ABB,1,10,01035B00,01035B02,01035B04,01035B14,01035B16,01035B18,01035B1A,r010350080004,r010350000004,r010350040004
1,010304UUuuUUuu@i0:10,Voltage L1-N,V,Voltage_L1,1
1,010304UUuuUUuu@i1:10,Voltage L2-N,V,Voltage_L2,1
1,010304UUuuUUuu@i2:10,Voltage L3-N,V,Voltage_L3,1
1,010304SSssSSss@i3:100,Active power Total,W,Active_power_Total,2
1,010304SSssSSss@i4:100,Active power L1-N,W,Active_power_L1,2
1,010304SSssSSss@i5:100,Active power L2-N,W,Active_power_L2,2
1,010304SSssSSss@i6:100,Active power L3-N,W,Active_power_L3,2
1,010308xxxxxxxxSSssSSss@i7:100,Real energy,kWh,Real_energy,2
1,010308xxxxxxxxUUuuUUuu@i8:100,Real energy consumed,kWh,Real_energy_consumed,2
1,010308xxxxxxxxUUuuUUuu@i9:100,Real energy delivered,kWh,Real_energy_delivered,2
1,020304UUuuUUuu@i0:10,Voltage L1-N,V,Voltage_L1,1
1,020304UUuuUUuu@i1:10,Voltage L2-N,V,Voltage_L2,1
1,020304UUuuUUuu@i2:10,Voltage L3-N,V,Voltage_L3,1
1,020304SSssSSss@i3:100,Active power Total,W,Active_power_Total,2
1,020304SSssSSss@i4:100,Active power L1-N,W,Active_power_L1,2
1,020304SSssSSss@i5:100,Active power L2-N,W,Active_power_L2,2
1,020304SSssSSss@i6:100,Active power L3-N,W,Active_power_L3,2
1,020308xxxxxxxxSSssSSss@i7:100,Real energy,kWh,Real_energy,2
1,020308xxxxxxxxUUuuUUuu@i8:100,Real energy consumed,kWh,Real_energy_consumed,2
1,020308xxxxxxxxUUuuUUuu@i9:100,Real energy delivered,kWh,Real_energy_delivered,2

**Comment by gemu2015** [comment]
example from the docs

```
>D
>B
->sensor53 r
>M 1
+1,3,M,1,9600,Meter,1,1,01030023,01030028,0103002d,01030025,0103002a,0103002f,01030032,01030027,0103002c,01030031,0103001B,0103001d,03030023,03030028,0303002d,03030025,0303002a,0303002f,03030032,03030027,0303002c,03030031,0303001B,0303001d
1,=h Domestic Electricity:
1,010304UUuuUUuu@i10:100,1 Tariff 1 total,kWh,M1_T1_total,2
1,010304UUuuUUuu@i11:100,1 Tariff 1 partial,kWh,M1_T1_par,2
1,=h Readings:
1,010304UUuu@i0:1,1 Voltage L1,V,M1_Voltage_L1,0
1,010304UUuu@i1:1,1 Voltage L2,V,M1_Voltage_L2,0
1,010304UUuu@i2:1,1 Voltage L3,V,M1_Voltage_L3,0
1,010304xxxxUUuu@i0:10,1 Current L1,A,M1_Current_L1,2
1,010304xxxxUUuu@i1:10,1 Current L2,A,M1_Current_L2,2
1,010304xxxxUUuu@i2:10,1 Current L3,A,M1_Current_L3,2
1,010304UUuu@i3:100,1 Active Power L1,kW,M1_PRMS_L1,3
1,010304UUuu@i4:100,1 Active Power L2,kW,M1_PRMS_L2,3
1,010304UUuu@i5:100,1 Active Power L3,kW,M1_PRMS_L3,3
1,010304UUuu@i6:100,1 Active Power total,kW,M1_PRMS_total,3
1,010304xxxxSSss@i3:100,1 Reactive Power L1,kVAr,M1_QRMS_L1,3
1,010304xxxxSSss@i4:100,1 Reactive Power L2,kVAr,M1_QRMS_L2,3
1,010304xxxxSSss@i5:100,1 Reactive Power L3,kVAr,M1_QRMS_L3,3
1,010304xxxxSSss@i6:100,1 Reactive Power total,kVAr,M1_QRMS_total,3
1,010304UUuu@i7:100,1 CosPhi L1,,M1_CosPhi_L1,2
1,010304UUuu@i8:100,1 CosPhi L2,,M1_CosPhi_L2,2
1,010304UUuu@i9:100,1 CosPhi L3,,M1_CosPhi_L3,2
1,=h________________________________________________
; meter 2 +12 offset
1,=h Heat Pump
1,030304UUuuUUuu@i22:100,2 Tariff 1 total,kWh,M2_T1_total,2
1,030304UUuuUUuu@i23:100,2 Tariff 1 partial,kWh,M2_T1_par,2
1,=h Readings:
1,030304UUuu@i12:1,2 Voltage L1,V,M2_Voltage_L1,0
1,030304UUuu@i13:1,2 Voltage L2,V,M2_Voltage_L2,0
1,030304UUuu@i14:1,2 Voltage L3,V,M2_Voltage_L3,0
1,030304xxxxUUuu@i12:10,2 Current L1,A,M2_Current_L1,2
1,030304xxxxUUuu@i13:10,2 Current L2,A,M2_Current_L2,2
1,030304xxxxUUuu@i14:10,2 Current L3,A,M2_Current_L3,2
1,030304UUuu@i15:100,2 Active Power L1,kW,M2_PRMS_L1,3
1,030304UUuu@i16:100,2 Active Power L2,kW,M2_PRMS_L2,3
1,030304UUuu@i17:100,2 Active Power L3,kW,M2_PRMS_L3,3
1,030304UUuu@i18:100,2 Active Power total,kW,M2_PRMS_total,3
1,030304xxxxSSss@i15:100,2 Reactive Power L1,kVAr,M2_QRMS_L1,3
1,030304xxxxSSss@i16:100,2 Reactive Power L2,kVAr,M2_QRMS_L2,3
1,030304xxxxSSss@i16:100,2 Reactive Power L3,kVAr,M2_QRMS_L3,3
1,030304xxxxSSss@i18:100,2 Reactive Power total,kVAr,M2_QRMS_total,3
1,030304UUuu@i19:100,2 CosPhi L1,,M2_CosPhi_L1,2
1,030304UUuu@i20:100,2 CosPhi L2,,M2_CosPhi_L2,2
1,030304UUuu@i21:100,2 CosPhi L3,,M2_CosPhi_L3,2
#
```

**Comment by spoeh-man** [comment]
Next Try im now on the right way?

D
B
->sensor53 r
M 1
+1,3,m,0,9600,ABB,1,10,01035B00,01035B02,01035B04,01035B14,01035B16,01035B18,01035B1A,r010350080004,r010350000004,r010350040004,02035B00,02035B02,02035B04,02035B14,02035B16,02035B18,02035B1A,r020350080004,r020350000004,r020350040004
1,=h Meter1:
1,010304UUuuUUuu@i0:10,Voltage L1-N,V,Voltage_L1,1
1,010304UUuuUUuu@i1:10,Voltage L2-N,V,Voltage_L2,1
1,010304UUuuUUuu@i2:10,Voltage L3-N,V,Voltage_L3,1
1,010304SSssSSss@i3:100,Active power Total,W,Active_power_Total,2
1,010304SSssSSss@i4:100,Active power L1-N,W,Active_power_L1,2
1,010304SSssSSss@i5:100,Active power L2-N,W,Active_power_L2,2
1,010304SSssSSss@i6:100,Active power L3-N,W,Active_power_L3,2
1,010308xxxxxxxxSSssSSss@i7:100,Real energy,kWh,Real_energy,2
1,010308xxxxxxxxUUuuUUuu@i8:100,Real energy consumed,kWh,Real_energy_consumed,2
1,010308xxxxxxxxUUuuUUuu@i9:100,Real energy delivered,kWh,Real_energy_delivered,2
1,=h Meter2:
1,020304UUuuUUuu@i0:10,Voltage L1-N,V,Voltage_L1,1
1,020304UUuuUUuu@i1:10,Voltage L2-N,V,Voltage_L2,1
1,020304UUuuUUuu@i2:10,Voltage L3-N,V,Voltage_L3,1
1,020304SSssSSss@i3:100,Active power Total,W,Active_power_Total,2
1,020304SSssSSss@i4:100,Active power L1-N,W,Active_power_L1,2
1,020304SSssSSss@i5:100,Active power L2-N,W,Active_power_L2,2
1,020304SSssSSss@i6:100,Active power L3-N,W,Active_power_L3,2
1,020308xxxxxxxxSSssSSss@i7:100,Real energy,kWh,Real_energy,2
1,020308xxxxxxxxUUuuUUuu@i8:100,Real energy consumed,kWh,Real_energy_consumed,2
1,020308xxxxxxxxUUuuUUuu@i9:100,Real energy delivered,kWh,Real_energy_delivered,2
#

**Reply by gemu2015** [reply]
Check the indices @0;@1
Each decode line needs an incremented index
But otherwise it looks ok

**Comment by spoeh-man** [comment, marked as answer]
Ok i think i have it for now

```
>D
>B
->sensor53 r
>M 1
+1,3,m,0,9600,ABB,1,10,01035B00,01035B02,01035B04,01035B14,01035B16,01035B18,01035B1A,r010350080004,r010350000004,r010350040004,02035B00,02035B02,02035B04,02035B14,02035B16,02035B18,02035B1A,r020350080004,r020350000004,r020350040004
1,=h Meter1:
1,010304UUuuUUuu@i0:10,Voltage L1-N,V,Voltage_L1,1
1,010304UUuuUUuu@i1:10,Voltage L2-N,V,Voltage_L2,1
1,010304UUuuUUuu@i2:10,Voltage L3-N,V,Voltage_L3,1
1,010304SSssSSss@i3:100,Active power Total,W,Active_power_Total,2
1,010304SSssSSss@i4:100,Active power L1-N,W,Active_power_L1,2
1,010304SSssSSss@i5:100,Active power L2-N,W,Active_power_L2,2
1,010304SSssSSss@i6:100,Active power L3-N,W,Active_power_L3,2
1,010308xxxxxxxxSSssSSss@i7:100,Real energy,kWh,Real_energy,2
1,010308xxxxxxxxUUuuUUuu@i8:100,Real energy consumed,kWh,Real_energy_consumed,2
1,010308xxxxxxxxUUuuUUuu@i9:100,Real energy delivered,kWh,Real_energy_delivered,2
1,=h Meter2:
1,020304UUuuUUuu@i10:10,Voltage L1-N,V,Voltage_L1,1
1,020304UUuuUUuu@i11:10,Voltage L2-N,V,Voltage_L2,1
1,020304UUuuUUuu@i12:10,Voltage L3-N,V,Voltage_L3,1
1,020304SSssSSss@i13:100,Active power Total,W,Active_power_Total,2
1,020304SSssSSss@i14:100,Active power L1-N,W,Active_power_L1,2
1,020304SSssSSss@i15:100,Active power L2-N,W,Active_power_L2,2
1,020304SSssSSss@i16:100,Active power L3-N,W,Active_power_L3,2
1,020308xxxxxxxxSSssSSss@i17:100,Real energy,kWh,Real_energy,2
1,020308xxxxxxxxUUuuUUuu@i18:100,Real energy consumed,kWh,Real_energy_consumed,2
1,020308xxxxxxxxUUuuUUuu@i19:100,Real energy delivered,kWh,Real_energy_delivered,2
#
```
